src/Adapter/Tornado/EventLoop.php

```diff
@@ -97,11 +107,26 @@ function (\Throwable $throwable) use ($task) {
      */
     public function async(\Generator $generator): Promise
     {
-        $this->tasks[] = ($task = new Internal\Task(
-            $generator,
-            Internal\PendingPromise::createUnhandled($this->unhandledFailingPromises))
+        /** @var Task $task */
+        $task = null;
+        /** @var PendingPromise $promise */
+        $promise = null;
+
+        $promise = Internal\PendingPromise::createUnhandled($this->unhandledFailingPromises,
+            function () use (&$task, &$promise) {
+                if ($task->getGenerator()->current()) {
+                    $promise->reject(new CancelledException('async cancellation'));
```

Comment:
This callback is called if the promise is canceled, and here you try to reject the canceled promise… not sure if it's relevant 🤔